File: array/532.py
class Solution:
    def findPairs(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        if not nums or len(nums) <= 1:
            return 0
        
        ret = 0
        nums.sort()
        i = 0
        while i < len(nums) - 1:
            pre = i
            nxt = i+1
            while nxt < len(nums) and nums[nxt] - nums[pre] <= k:
                if nums[nxt] - nums[pre] == k:
                    ret += 1
                    while i < len(nums)-1 and nums[i] == nums[i+1]:
                        i += 1
                    break
                else:
                    nxt += 1
            i += 1
        
        return ret
                        
        
        

File: array/532.cc
class Solution {
public:
    int findPairs(vector<int>& nums, int k) {
        int ret = 0;
        std::unordered_map<int, int> m_;
        for (auto& num : nums) {
            auto iter = m_.find(num);
            if (iter != m_.end()) {
                (iter->second)++;
            } else {
                m_.emplace(num, 1);
            }
        }
        for (auto iter = m_.begin(); iter != m_.end(); ++iter) {
            auto tar_iter = m_.find(iter->first+k);
            if (tar_iter != m_.end()) {
                if (k == 0) {
                    if (tar_iter->second >= 2)
                        ret++;
                } else {
                    ret++;    
                }
            }
        }
        return ret;
    }
    
};
